src/renderer/settings/components/GeneralSettings/index.js

```diff
@@ -15,5 +20,15 @@ const mapCurrencyActionsToProps = (actions) => ({
 export default compose(
   withActions(setCurrency, mapCurrencyActionsToProps),
   withData(currencyActions, mapCurrencyDataToProps),
-  withReset(pricesActions, 'currency')
+  withReset(pricesActions, 'currency'),
+
+  withSuccessToast('showSuccessToast'),
+  withProgressChange(setCurrency, LOADED, (state, props) => {
+    props.showSuccessToast('Settings successfully updated');
+  }),
+
+  withErrorToast('showErrorToast'),
```

Comment:
can't this be an enum, like you've used in the reducer? `TYPE_ERROR `or imply the `showErrorToast` by `withErrorToast`